package/MDAnalysis/lib/distances.py

```diff
@@ -367,75 +362,82 @@ def self_distance_array(reference, box=None, result=None, backend="serial"):
+    """Calculates pairs of indices corresponding to entries in the `reference`
+    and `configuration` arrays which are separated by a distance lying within
+    the specified cutoff(s). Optionally, these distances can be returned as
+    well.
 
-    If a *box* is supplied, then a minimum image convention is used
-    to evaluate the distances.
+    If the optional argument `box` is supplied, the minimum image convention is
+    applied when calculating distances. Either orthogonal or triclinic boxes are
+    supported.
 
-    An automatic guessing of optimized method to calculate the distances is
+    An automatic guessing of the optimal method to calculate the distances is
     included in the function. An optional keyword for the method is also
-    provided. Users can override the method with this functionality.
-    Currently pkdtree and bruteforce are implemented.
-
+    provided. Users can enforce a particular method with this functionality.
+    Currently brute force, grid search, and periodic KDtree methods are
+    implemented.
 
     Parameters
     -----------
-    reference : array
-        reference coordinates array with shape ``reference.shape = (3,)``
-        or ``reference.shape = (len(reference), 3)``
-    configuration : array
-        Configuration coordinate array with shape ``reference.shape = (3,)``
-        or ``reference.shape = (len(reference), 3)``
+    reference : numpy.ndarray
+        Reference coordinate array with shape ``(3,)`` or ``(n, 3)``.
+    configuration : numpy.ndarray
+        Configuration coordinate array with shape ``(3,)`` or ``(m, 3)``.
     max_cutoff : float
-        Maximum cutoff distance between the reference and configuration
-    min_cutoff : (optional) float
-        Minimum cutoff distance between reference and configuration [None]
-    box : (optional) array or None
-        The dimensions, if provided, must be provided in the same
-        The unitcell dimesions for this system format as returned
-        by :attr:`MDAnalysis.coordinates.base.Timestep.dimensions`:
-        ``[lx,ly, lz, alpha, beta, gamma]``. Minimum image convention
-        is applied if the box is provided [None]
-    method : (optional) 'bruteforce' or 'pkdtree' or 'None'
-        Keyword to override the automatic guessing of method built-in
-        in the function [None]
-    return_distances : (optional) 'True'/'False'
-        Function returns distances if set to 'True'
+        Maximum cutoff distance between the reference and configuration.
+    min_cutoff : float, optional
+        Minimum cutoff distance between reference and configuration.
+    box : array_like, optional
+        The unitcell dimensions of the system, which can be orthogonal or
+        triclinic and must be provided in the same format as returned by
+        :attr:`MDAnalysis.coordinates.base.Timestep.dimensions`:\n
+        ``[lx, ly, lz, alpha, beta, gamma]``.
+    method : str, optional
```

Comment:
You can provide the selection with
```reST
method : {"bruteforce", "nsgrid", "pkdtree", None}
     Keyword to override the automatic guessing ofthe employed search method.
     ``None`` chooses an optimal method.
```
(see https://numpydoc.readthedocs.io/en/latest/format.html#sections under _4. Parameters_)

Reply:
The numpydoc docstring guide also says
> If it is not necessary to specify a keyword argument, use `optional`.

And furthermore (what you referred to)
> When a parameter can only assume one of a fixed set of values, those values can be listed in braces, with the default appearing first.

Listing the default `None` (which should come first, btw) doesn't really make sense to me, that would be like describing an optional kwarg as, for example, `x : int or None`.

I think a proper way of combining the two guidelines should look like this:
```
method : {'bruteforce', 'nsgrid', 'pkdtree'}, optional
    Keyword to override the automatic guessing of the employed search method.
```
The default is given in the function signature anyway.

What do you think?